### DISCUSSION TITLE
Vscodium issue

### DISCUSSION THREAD

**Discussion by DanielRein-1** [VSCodium/vscodium#1972] [Q&A]
am on linux using vscodium when i run files that have math.h i get an error how can i solve this to link my code with the math library

**Comment by Jeong-ilKajokaya** [comment]
Hi! The issue you’re encountering is common when compiling C/C++ code that uses `math.h`. Functions like `sin()`, `cos()`, `sqrt()`, etc., are part of the **math library**, which is separate and needs to be linked explicitly.

### How to fix it

#### 1. Using `gcc` or `g++` from the terminal

Add `-lm` at the end of your compile command:

```bash
gcc myfile.c -o myprogram -lm
```

or for C++:

```bash
g++ myfile.cpp -o myprogram -lm
```

* `-l` tells the linker to link a library.
* `m` is the math library (`libm`).

#### 2. If using tasks in VSCodium

If you are building via a task or using the built-in terminal, make sure the compile command includes `-lm`.

Example `tasks.json` snippet:

```json
{
  "label": "build math program",
  "type": "shell",
  "command": "gcc",
  "args": [
    "${file}",
    "-o",
    "${fileBasenameNoExtension}",
    "-lm"
  ],
  "group": {
    "kind": "build",
    "isDefault": true
  }
}
```

#### 3. Common mistakes

* Placing `-lm` **before** the source file may fail on some systems. Always place `-lm` **at the end**.
* Ensure you actually have the math library installed (on Linux it usually comes with `glibc`, so it’s normally available).

**Summary:**

* Add `-lm` when compiling.
* Make sure it comes after your source files in the command.
* This will correctly link the math library and resolve `math.h` function errors.

**Comment by DanielRein-1** [comment]
Thank you bro but i fixed it while ago

**Reply by Jeong-ilKajokaya** [reply]
My bad, I was just trying to work on unanswered questions in the discussion. 

I am always here to help 

![giphy](https://media.giphy.com/media/LllA2dKt1qZuE/giphy.gif?cid=790b7611cc616a7c9fe54b03d87fd9498119ef3632b52c32&ep=v1_user_favorites&rid=giphy.gif&ct=g)
